src/runtime_src/core/tools/common/reports/ReportTelemetry.cpp

```diff
@@ -0,0 +1,213 @@
+
+  return ss.str();
+}
+
+static std::string
+generate_opcode_string(const boost::property_tree::ptree& pt)
+{
+  std::stringstream ss;
+
+  const std::vector<Table2D::HeaderData> opcode_headers = {
+    {"Opcode", Table2D::Justification::left},
+    {"Receive Count", Table2D::Justification::left}
+  };
+  Table2D opcode_table(opcode_headers);
+
+  int index = 0;
+  boost::property_tree::ptree opcodes = pt.get_child("opcodes", empty_ptree);
+  for (const auto& [name, opcode] : opcodes) {
+    std::vector<std::string> opcode_data = {
+      std::to_string(index),
+      std::to_string(opcode.get<uint64_t>("received_count"))
+    };
+    opcode_table.addEntry(opcode_data);
+    index++;
+  }
+
+  ss << opcode_table.toString("  ");
+
+  return ss.str();
+}
+
+static std::string
+generate_stream_buffer_string(const boost::property_tree::ptree& pt)
+{
+  std::stringstream ss;
+
+  const std::vector<Table2D::HeaderData> stream_buffer_headers = {
+    {"Stream Buffer", Table2D::Justification::left},
+    {"Tokens", Table2D::Justification::left}
+  };
+  Table2D stream_buffer_table(stream_buffer_headers);
+
+  int index = 0;
+  boost::property_tree::ptree stream_buffers = pt.get_child("stream_buffers", empty_ptree);
+  for (const auto& [name, stream_buffer] : stream_buffers) {
+    std::vector<std::string> stream_buffer_data = {
+      std::to_string(index),
+      std::to_string(stream_buffer.get<uint64_t>("tokens"))
+    };
+    stream_buffer_table.addEntry(stream_buffer_data);
+    index++;
+  }
+
+  ss << stream_buffer_table.toString("  ");
+
+  return ss.str();
+}
+
+static std::string
+generate_aie_string(const boost::property_tree::ptree& pt)
+{
+  std::stringstream ss;
+
+  const std::vector<Table2D::HeaderData> aie_headers = {
+    {"AIE Column", Table2D::Justification::left},
+    {"Deep Sleep Count", Table2D::Justification::left}
+  };
+  Table2D aie_table(aie_headers);
+
+  int index = 0;
+  boost::property_tree::ptree aie_cols = pt.get_child("aie_columns", empty_ptree);
+  for (const auto& [name, aie_col] : aie_cols) {
+    std::vector<std::string> aie_data = {
+      std::to_string(index),
+      std::to_string(aie_col.get<uint64_t>("deep_sleep_count"))
+    };
+    aie_table.addEntry(aie_data);
+    index++;
+  }
+
+  ss << aie_table.toString("  ");
+
+  return ss.str();
+}
+
+static std::string
+generate_misc_string(const boost::property_tree::ptree& pt)
+{
+  std::stringstream ss;
+
+  ss << boost::format("  %-23s: %s \n") % "L1 Interrupt Count" % pt.get<std::string>("level_one_interrupt_count");
+
+  return ss.str();
+}
+
+void
+ReportTelemetry::writeReport(const xrt_core::device* /*_pDevice*/,
+                             const boost::property_tree::ptree& pt,
+                             const std::vector<std::string>& /*_elementsFilter*/,
+                             std::ostream& _output) const
+{
+  _output << "Telemetry\n";
+
+  boost::property_tree::ptree telemetry_pt = pt.get_child("telemetry");
```

Comment:
Just add the "info not available" but the rest looks good. Thanks for the comments in the query_requests!